src/App.js

```diff
@@ -8,14 +9,28 @@ import { Todolist } from "./components/Todolist";
 import { Wishlist } from "./components/Wishlist";
 import { AddForm } from "./components/AddForm";
 import { Help } from "./components/Help";
+import { LoginSite } from "./components/LoginSite";
+import { RegistrationSite } from "./components/RegistrationSite";
 import { Error } from "./components/Error";
 import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
 import { PlantView } from "./components/PlantView";
-import { ImageUpload } from "./components/ImageUpload";
-import { ImagesGallery } from "./components/ImagesGallery";
-import { SinglePlantTimeline } from "./components/SinglePlantTimeline";
 
 export class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: null
+    };
+  }
+
+  componentDidMount = () => {
+    firebase.auth().onAuthStateChanged(user => {
+      if (user) {
```

Comment:
I think we do not need the `if` statement here. If you log out you probably want to set user back to `null`